src/dreyfus_index_updater.erl

```diff
@@ -17,20 +17,52 @@
 -include_lib("couch/include/couch_db.hrl").
 -include("dreyfus.hrl").
 
--export([update/2, load_docs/3]).
+-export([update/2, load_docs/2]).
 
 -import(couch_query_servers, [get_os_process/1, ret_os_process/1, proc_prompt/2]).
 
 update(IndexPid, Index) ->
     #index{
         current_seq = CurSeq,
+        purge_seq = IdxPurgeSeq,
         dbname = DbName,
         ddoc_id = DDocId,
         name = IndexName
     } = Index,
     erlang:put(io_priority, {view_update, DbName, IndexName}),
     {ok, Db} = couch_db:open_int(DbName, []),
     try
+        {ok, DbPurgeSeq} = couch_db:get_purge_seq(Db),
+        if IdxPurgeSeq == DbPurgeSeq ->
+            NewPurgeSeq = IdxPurgeSeq;
+        true ->
+            {ok, DbOldestPurgeSeq} = couch_db:get_oldest_purge_seq(Db),
+
+            FoldFun = fun(PurgeSeq, {Id, Revs}, Acc) ->
+                {ok, [ {PurgeSeq, Id, Revs} | Acc]}
+            end,
+            if (IdxPurgeSeq + 1) >= DbOldestPurgeSeq ->
+                {ok, PurgeSeqIdRevs} = couch_db:fold_purged_docs(Db, IdxPurgeSeq, FoldFun, [], []),
+
+                PSeqs = lists:foldr(fun({PSeq, Id, _}, Acc) ->
+                    clouseau_rpc:delete(IndexPid, Id),
+                    clouseau_rpc:set_purge_seq(IndexPid, PSeq),
+                    [PSeq | Acc]
+                end, [], PurgeSeqIdRevs),
```

Comment:
Instead of list for Acc you can use just a starting purge_seq, so at the end you will get your final purge_seq for the accumulator, and you don't need to do `lists:max(PSeqs)` on line 56.